framework/src/play/classloading/ApplicationClassloader.java

```diff
@@ -470,11 +468,13 @@ public int compare(Class o1, Class o2) {
     public Class getClassIgnoreCase(String name) {
         getAllClasses();
         for (ApplicationClass c : Play.classes.all()) {
-            if (c.name.equalsIgnoreCase(name) || c.name.replace("$", ".").equalsIgnoreCase(name)) {
-                if (Play.usePrecompiled) {
-                    return c.javaClass;
-                }
-                return loadApplicationClass(c.name);
+            if (c.name.equalsIgnoreCase(name)) {
+                return Play.usePrecompiled ? c.javaClass : loadApplicationClass(c.name);
+            }
+        }
+        for (ApplicationClass c : Play.classes.all()) {
```

Comment:
@tazmaniax @xael-fry Sorry for ignoring your question.

The goal is to improve performance. Operations `equalsIgnoreCase` and especially `.replace("$", ".")` are relatively slow - they are executed many times in the loop. But in most cases it's not needed, because developers do write correct class names in HTML (like `@{MyController.myAction()}`).

So, now the algorithm is:
1. Try the **fastest method** that covers most of cases - `loadApplicationClass(name);`   (take class from Map, avoid looping)
2. Try the second **slower method** in loop - `c.name.equalsIgnoreCase(name)`
3. Try the **slowest method** `c.name.replace("$", ".")`.